src/Features/CSharpTest/ExtractMethod/ExtractMethodCodeRefactoringTests.cs

```diff
@@ -6079,4 +6085,3011 @@ private static int NewMethod(int i, int j)
+                private int Repro(int[] x)
+                {
+                    foreach (var v in x)
+                    {
+                        bool flowControl = {|Rename:NewMethod|}(v);
+                        if (flowControl)
+                        {
+                            continue;
+                        }
+                        else
+                        {
+                            break;
+                        }
+                    }
+            
+                    return 0;
+                }
+
+                private static bool NewMethod(int v)
+                {
+                    if (v == 0)
+                    {
+                        return true;
+                    }
+
+                    return false;
+                }
+            }
+            """);
+    }
+
+    [Fact]
+    public async Task TestFlowControl_ContinueAndContinue()
+    {
+        await TestInRegularAndScript1Async(
+            """
+            class C
+            {
+                private int Repro(int[] x)
+                {
+                    foreach (var v in x)
+                    {
+                        [|if (v == 0)
+                        {
+                            continue;
+                        }
+                        
+                        continue;|]
+                    }
+
+                    return 0;
+                }
+            }
+            """,
+            """
+            class C
+            {
+                private int Repro(int[] x)
+                {
+                    foreach (var v in x)
+                    {
+                        {|Rename:NewMethod|}(v);
+                        continue;
+                    }
+            
+                    return 0;
+                }
+
+                private static void NewMethod(int v)
+                {
+                    if (v == 0)
+                    {
+                        return;
+                    }
+
+                    return;
+                }
+            }
+            """);
+    }
+
+    [Fact]
+    public async Task TestFlowControl_ContinueAndReturn()
+    {
+        await TestInRegularAndScript1Async(
+            """
+            class C
+            {
+                private int Repro(int[] x)
+                {
+                    foreach (var v in x)
+                    {
+                        [|if (v == 0)
+                        {
+                            continue;
+                        }
+                        
+                        return 1;|]
+                    }
+
+                    return 0;
+                }
+            }
+            """,
+            """
+            class C
+            {
+                private int Repro(int[] x)
+                {
+                    foreach (var v in x)
+                    {
+                        (bool flowControl, int value) = {|Rename:NewMethod|}(v);
+                        switch (flowControl)
```

Comment:
Talked offline and the use of a `switch` here is for conciseness. I do wonder if it will be what developers expect and how often it will be rewritten to an `if else`.

Reply:
my personal preference would be to use switches 100% of the time.  the problem is that we can't use that for 'break' as then 'break' would literally just break us out of the switch we are inserting :'(

i don't think of this as a set of values we check one at a time.  i think of this as a literlal value that jumps us to the exact control flow construct we want.